lib/fs.js

```diff
@@ -1818,7 +1826,7 @@ StatWatcher.prototype.stop = function() {
 const statWatchers = new Map();
 
 fs.watchFile = function(filename, options, listener) {
-  handleError((filename = getPathFromURL(filename)));
+  handleError((filename = getPathFromURL(filename, true)));
```

Comment:
We actually support `fs.watchFile` and `fs.unwatchFile` on directories. There are even tests in the async-hook test suites.